src/game/objects/bridge.c

```diff
@@ -103,14 +114,31 @@ void DrawBridgeCollision(
 void BridgeFlatFloor(
     ITEM_INFO *item, int32_t x, int32_t y, int32_t z, int16_t *height)
 {
-    if (y <= item->pos.y) {
+    if (g_Config.fix_bridge_grabs) {
+        if (!isSameSector(x, y, z, item)) {
+            return;
+        }
+        if (y > item->pos.y) {
+            return;
+        }
+
         *height = item->pos.y;
+    } else {
+        if (y <= item->pos.y) {
+            *height = item->pos.y;
+        }
     }
```

Comment:
Current logic is hard to follow and should be simplified to something like

```c
if (g_Config.fix_bridge_grabs && !Bridge_IsSameSector(x, y, z, item)) {
    return;
}
if (y > item->pos.y) {
    return;
}
*height = item->pos.y;
```

Reply:
Great point. Looks much cleaner and easier to follow.